include/extractor/guidance/turn_handler.hpp

```diff
@@ -45,9 +47,24 @@ class TurnHandler : public IntersectionHandler
                             Intersection intersection) const override final;
 
   private:
+    struct Fork
+    {
+        Intersection::iterator right;
+        Intersection::iterator left;
+        const std::size_t size;
+        Fork(Intersection::iterator right, Intersection::iterator left);
+    };
+
     bool isObviousOfTwo(const EdgeID via_edge,
                         const ConnectedRoad &road,
                         const ConnectedRoad &other) const;
+
+    bool hasObvious(const EdgeID &via_edge, const Fork) const;
+
+    boost::optional<Fork> findLeftAndRightmostForkCandidates(Intersection &intersection) const;
+
+    bool isCompatibleByRoadClass(const Intersection &intersection, const Fork fork) const;
```

Comment:
should probably move to a dedicated `.compatible()` in `road_class`? We can do so in refactoring road classes, though.